src/main/flight/pid.c

```diff
@@ -1220,14 +1250,17 @@ void FAST_CODE pidController(const pidProfile_t *pidProfile, timeUs_t currentTim
             // yaw has a tendency to windup. Otherwise limit yaw iterm accumulation.
             const int launchControlYawItermLimit = (pidRuntime.launchControlMode == LAUNCH_CONTROL_MODE_FULL) ? LAUNCH_CONTROL_YAW_ITERM_LIMIT : 0;
             pidData[FD_YAW].I = constrainf(pidData[FD_YAW].I, -launchControlYawItermLimit, launchControlYawItermLimit);
+            pidData[FD_YAW].I_afterTpa = pidData[FD_YAW].I;
 
             // for pitch-only mode we disable everything except pitch P/I
             if (pidRuntime.launchControlMode == LAUNCH_CONTROL_MODE_PITCHONLY) {
                 pidData[FD_ROLL].P = 0;
                 pidData[FD_ROLL].I = 0;
+                pidData[FD_ROLL].I_afterTpa = 0;
                 pidData[FD_YAW].P = 0;
                 // don't let I go negative (pitch backwards) as front motors are limited in the mixer
                 pidData[FD_PITCH].I = MAX(0.0f, pidData[FD_PITCH].I);
+                pidData[FD_PITCH].I_afterTpa = pidData[FD_PITCH].I;
```

Comment:
Poor code structure manifests in this way...

Reply:
I don't see how I can make this PR the other way...

Reply:
Sorry, just a comment. It is problem with PID code, not this PR ...

Reply:
> Sorry, just a comment. It is problem with PID code, not this PR ...

could you please trigger a rebuild for the PR? I don't have privileges. AT32 targets failed, i don't see why.

Reply:
It seems to fail on resource limit. I did restart it once and it failed again. But I don't understand github CI

Reply:
Seems good now! thanks

Reply:
i triggered it again and enabled debugging, but it succeeded, so nothing to debug i suppose.